### PR TITLE
Check for null in Project.load()

### PR DESCRIPTION
**Pull request by ajkannan**
From discussion in #516

### CODE REVIEW

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java [hunk 1]
```diff
@@ -42,12 +42,13 @@ public Project(ResourceManager resourceManager, ProjectInfo projectInfo) {
   /**
    * Constructs a Project object that contains project information loaded from the server.
    *
-   * @return Project object containing the project's metadata
+   * @return Project object containing the project's metadata or
+   *     null if the project doesn't exist
```
Comment (mziccard) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java]: Nit, couldn't null fit on the previous line? Also we better use `{@code null}`

File: gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java [hunk 2]
```diff
@@ -67,7 +68,8 @@ public ResourceManager resourceManager() {
   /**
    * Returns a Project object with updated project information.
    *
-   * @return Project object containing the project's updated metadata
+   * @return Project object containing the project's updated metadata or null if the project no
+   *     longer exists
```
Comment (mziccard) [inline gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java]: `null` could also be returned if the user feeds to the Project class infos of a non-existing project, this is unlikely to happen but I was thinking about phrasing this as: `or {@code null} if not found`

**Comment by ajkannan** [comment]
@mziccard fixed in place

**Comment by mziccard** [comment]
@ajkannan What do you think about adding tests for `Project`'s `load` and `reload` for the cases when the service returns `null`? Something like:
```java
@Test
public void testLoadNull() {
  expect(resourceManager.get(PROJECT_INFO.projectId())).andReturn(null);
  replay(resourceManager);
  assertNull(Project.load(resourceManager, PROJECT_INFO.projectId()));
}
```

**Comment by ajkannan** [comment]
I added a test for `reload` already, and I think that covers `load` since `reload` uses `load`.  But I can still add a test like the one you listed in case `reload` doesn't use `load` in the future.

**Comment by ajkannan** [comment]
@mziccard Added the `testLoadNull` test

**Comment by mziccard** [comment]
Looks good to me!
